pkg/controller/volume/persistentvolume/scheduler_binder.go

```diff
@@ -141,6 +141,27 @@ func (b *volumeBinder) GetBindingsCache() PodBindingCache {
 
 // FindPodVolumes caches the matching PVs and PVCs to provision per node in podBindingCache
 func (b *volumeBinder) FindPodVolumes(pod *v1.Pod, node *v1.Node) (unboundVolumesSatisfied, boundVolumesSatisfied bool, err error) {
+	var (
+		matchedClaims     []*bindingInfo
+		provisionedClaims []*v1.PersistentVolumeClaim
+	)
+	defer func() {
+		// We recreate bindings for each new schedule loop.
+		// Although we do not distinguish nil from empty in this function, for
+		// easier testing, we normalize empty to nil.
+		if len(matchedClaims) == 0 {
+			matchedClaims = nil
+		}
+		if len(provisionedClaims) == 0 {
+			provisionedClaims = nil
+		}
+		// TODO merge into one atomic function
```

Comment:
In current implementation, IMO there is no need to separate `bindings` and `provisionedPVCs` in pod binding cache. It's better to get/update/delete `bindings` and `provisionedPVCs` together.